tools/stimulus.py

```diff
@@ -1,160 +1,178 @@
         else:
-            self.tokens.append( ("text", line[0].strip()) )
+            self.tokens.append(("text", line[0].strip()))
             for k in keys:
-                self.tokens.append( ("dedent", "") )
-                self.tokens.append( ("key", k) )
-                self.tokens.append( ("indent", "") )
-            keys=[]
-                
+                self.tokens.append(("dedent", ""))
+                self.tokens.append(("key", k))
+                self.tokens.append(("indent", ""))
+            keys = []
+
         if self.tokens:
             return self.tokens.pop(0)
 
+
 ################################################################################
 class PParser:
     def parse(self, stream):
-        lex=PLexer(stream)
-        val=seqdict.seqdict()
-        val_stack=[val, None]
-        nam_stack=[None, None]
-        
-        tok=lex.token()
-        while not tok[0]=="eof":
-            if tok[0]=="indent":
+        lex = PLexer(stream)
+        val = OrderedDict()
+        val_stack = [val, None]
+        nam_stack = [None, None]
+
+        tok = lex.token()
+        while not tok[0] == "eof":
+            if tok[0] == "indent":
                 val_stack.append(None)
                 nam_stack.append(None)
-            elif tok[0]=="dedent":
-                v=val_stack.pop()
-                n=nam_stack.pop()
-                if n is None:                    
-                    val_stack[-1]=v
+            elif tok[0] == "dedent":
+                v = val_stack.pop()
+                n = nam_stack.pop()
+                if n is None:
+                    val_stack[-1] = v
                 else:
-                    val_stack[-1][n]=v
-            elif tok[0]=="key":
+                    val_stack[-1][n] = v
+            elif tok[0] == "key":
                 if not nam_stack[-1] is None:
-                    val_stack[-2][nam_stack[-1]]=val_stack[-1]
-                if tok[1][-5:]=="-list":
-                    val_stack[-1]=seqdict.seqdict()
-                    nam_stack[-1]=tok[1][:-5]
+                    val_stack[-2][nam_stack[-1]] = val_stack[-1]
+                if tok[1][-5:] == "-list":
+                    val_stack[-1] = OrderedDict()
+                    nam_stack[-1] = tok[1][:-5]
                 else:
-                    val_stack[-1]={}
-                    nam_stack[-1]=tok[1]
-            elif tok[0]=="text":
-                val_stack[-1]=tok[1]
-            tok=lex.token()
+                    val_stack[-1] = {}
+                    nam_stack[-1] = tok[1]
+            elif tok[0] == "text":
+                val_stack[-1] = tok[1]
+            tok = lex.token()
 
         return val
 
+
 ################################################################################
 
+
 def main():
     # Command line arguments
     try:
-        optlist, args = getopt.getopt(sys.argv[1:], 't:f:l:i:o:hv', ['help'])
+        optlist, args = getopt.getopt(sys.argv[1:], "t:f:l:i:o:hv", ["help"])
```

Comment:
`getopt` is obsolete, most Python code uses `argparse` nowadays, but this is not a priority.  We can do it any time in the future if we have some time.